frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/backgroundables/RenameBackgroundable.java

```diff
@@ -78,6 +84,57 @@ public void run(ProgressIndicator indicator) {
     // (Stepping deeper is not an option since we would lose some important logic or become very dependent on the internals of git4idea).
     // Hence, we wait for the creation of the branch (with exponential backoff).
     waitForCreationOfLocalBranch(gitRepository, newBranchName);
+
+    if (gitNewBranchOptions.shouldSetTracking()) {
```

Comment:
Hmmm

<img width="414" alt="image" src="https://user-images.githubusercontent.com/3383210/216782607-6629944c-83f0-47c4-8622-73892298f9dc.png">

1. `~/git-machete/graphics/setup-sandbox`
2. open IntelliJ on `~/machete-sandbox`
3. Rename `allow-ownership-link`
4. Whoops it's untracked now

Is it expected? 🤔

Reply:
Also:
<img width="913" alt="image" src="https://user-images.githubusercontent.com/3383210/216782667-c991c0ea-1469-4489-890c-cd3b4b44b439.png">
No trace of `git branch --set-upstream-to` in history 🤔

Reply:
you have not specified whether the set tracking branch checkbox is selected in your scenario.

if it is selected it is expected IMHO - we "rename" the remote branch to smth that does not exist in the remote repo yet 

in the other case the status to remote shall remain ofc ( and that's how it works to me )

Reply:
Huh okay... makes sense. Still, pretty surprising at first glance...